// ngx-fullpage/components/fullpage/mnFullpage.directive.spec.js
"use strict";
/**
 * @author Meiblorn (Vadim Fedorenko) <meiblorn@gmail.com | admin@meiblorn.com> on 15/05/16.
 */
Object.defineProperty(exports, "__esModule", { value: true });
var testing_1 = require("@angular/core/testing");
var core_1 = require("@angular/core");
var mnFullpage_directive_1 = require("./mnFullpage.directive");
var mnFullpage_options_class_1 = require("./mnFullpage-options.class");
describe('mnFullpage directive spec', function () {
    var TestComponent = (function () {
        function TestComponent() {
            this.options = new mnFullpage_options_class_1.MnFullpageOptions({
                controlArrows: false,
                scrollingSpeed: 1000,
                css3: true
            });
        }
        return TestComponent;
    }());
    TestComponent.decorators = [
        { type: core_1.Component, args: [{
                    template: "\n            <!--suppress HtmlUnknownAttribute -->\n            <div mnFullpage>Content</div>\n        ",
                },] },
    ];
    /** @nocollapse */
    TestComponent.ctorParameters = function () { return []; };
    TestComponent.propDecorators = {
        'options': [{ type: core_1.Input },],
    };
    beforeEach(function () {
        testing_1.TestBed.configureTestingModule({
            declarations: [
                mnFullpage_directive_1.MnFullpageDirective,
                TestComponent
            ]
        });
    });
    it('compiled element should have fullpage-wrapper class, 100% height style', testing_1.fakeAsync(function () {
        testing_1.TestBed.compileComponents().then(function () {
            var fixture = testing_1.TestBed.createComponent(TestComponent);
            fixture.detectChanges();
            testing_1.tick();
            var compiled = fixture.debugElement.nativeElement.children[0];
            expect(compiled.classList.contains('fullpage-wrapper')).toBe(true);
            expect(compiled.style.height).toBe('100%');
        });
    }));
});
